src/app/items/services/result-fetching.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { Item } from 'src/app/data-access/get-item-by-id.service';
+import { FullItemRoute, isRouteWithSelfAttempt } from 'src/app/models/routing/item-route';
+import { GetResultsService, Result } from '../data-access/get-results.service';
+import { EMPTY, Observable, map, of, switchMap } from 'rxjs';
+import { canCurrentUserViewContent } from '../models/item-view-permission';
+import { bestAttemptFromResults, implicitResultStart } from '../models/attempts';
+import { ResultActionsService } from 'src/app/data-access/result-actions.service';
+
+export function canFetchResults(item: Item): boolean {
+  return canCurrentUserViewContent(item);
+}
```

Comment:
For better naming or suppose to be extended ?

Reply:
It is just that for the external point of view, we just want to know if we "can fetch results". 
And basically, it is "computed" (currently at least) as  whether the current user can view content. 
If I use `canCurrentUserViewContent` directly in the selector, the reader cannot understand why I use this function.